packages/plugins/rubygem-llhttp-ffi/rubygem-llhttp-ffi.spec

```
@@ -0,0 +1,88 @@
+# template: default
+%global gem_name llhttp-ffi
+%global gem_require_name %{gem_name}
+
+Name: rubygem-%{gem_name}
+Version: 0.4.0
+Release: 1%{?dist}
+Summary: Ruby FFI bindings for llhttp
+License: MPL-2.0
+URL: https://github.com/metabahn/llhttp/
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
+
+# start specfile generated dependencies
+Requires: ruby >= 2.5.0
+BuildRequires: ruby-devel >= 2.5.0
+BuildRequires: rubygems-devel
+BuildRequires: rubygem(ffi-compiler) >= 1.0
+BuildRequires: rubygem(ffi-compiler) < 2
+BuildRequires: rubygem(rake) >= 13.0
+BuildRequires: rubygem(rake) < 14
+# Compiler is required for build of gem binary extension.
+# https://fedoraproject.org/wiki/Packaging:C_and_C++#BuildRequires_and_Requires
+BuildRequires: gcc
+# end specfile generated dependencies
+
+%description
+Ruby FFI bindings for llhttp.
+
+
+%package doc
+Summary: Documentation for %{name}
+Requires: %{name} = %{version}-%{release}
+BuildArch: noarch
+
+%description doc
+Documentation for %{name}.
+
+%prep
+%setup -q -n  %{gem_name}-%{version}
+
+%build
+# Create the gem as gem install only works on a gem file
+gem build ../%{gem_name}-%{version}.gemspec
+
+# %%gem_install compiles any C extensions and installs the gem into ./%%gem_dir
+# by default, so that we can move it into the buildroot in %%install
+%gem_install
+
+%install
+mkdir -p %{buildroot}%{gem_dir}
+cp -a .%{gem_dir}/* \
+        %{buildroot}%{gem_dir}/
+
+mkdir -p %{buildroot}%{gem_extdir_mri}/%{gem_name}
+cp -a .%{gem_extdir_mri}/gem.build_complete %{buildroot}%{gem_extdir_mri}/
+cp -a .%{gem_extdir_mri}/%{gem_name}/*.so %{buildroot}%{gem_extdir_mri}/%{gem_name}
```

I'm not exactly sure why it's so inconsistent, but sometimes the .so file is in a directory, sometimes it's not. I think this is one example where it's not. If you grep for `gem_extdir_mri` you'll see there's not a clear preference for either one or the other.

This may work:

```suggestion
cp -a .%{gem_extdir_mri}/{gem.build_complete,*.so} %{buildroot}%{gem_extdir_mri}/
```

---

What worked as path for me:

```
cp -a %{buildroot}%{gem_instdir}/ext/x86_64-linux/*.so %{buildroot}%{gem_extdir_mri}/%{gem_name}
```

But then the tests fail:

```
+ ruby -e 'require '\''llhttp-ffi'\'''
/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:83:in `require': cannot load such file -- llhttp-ffi (LoadError)
	from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:83:in `require'
	from -e:1:in `<main>'
```

---

https://www.rubydoc.info/gems/llhttp-ffi/0.4.0 suggests the name that you need to require `llhttp`. Our template doesn't really cover that so it needs some additional work.

---

I've changed the line as @dgoetz suggested. The build seems to work.

I've also changed the check line to match the resulting gem name (`llhttp`) - however, the check still fails. I'm wondering if the `*.so` file is indeed included and moved to the correct location?